### PR TITLE
MAt-376: Add function to create new donation based on regular giving mandate & donor account

### PR DESCRIPTION
**Pull request by bdsl**

### CODE REVIEW

File: src/Domain/Donation.php [hunk 1]
```diff
@@ -346,17 +346,48 @@ public static function emptyTestDonation(
...
     }
 
     /**
      * @psalm-param numeric-string $amount
+     * @psalm-param ?numeric-string $tipAmount
+     * @psalm-param Campaign $campaign
      */
-    private function __construct(string $amount, string $currencyCode, PaymentMethodType $paymentMethodType)
-    {
+    public function __construct(
```
Comment (bdsl) [inline src/Domain/Donation.php:377]: refactored to make this public so I could call it from the Mandate class. In the process moved code into this function from `fromApiModel` which now pretty much just unpacks the arguments.

**Review by NoelLH — COMMENTED** [review]
All LGTM, except the twin migrations which I think might be committed together in error?

**Review by NoelLH — APPROVED** [review]
Missed that the later migration has the `DROP`s needed in its `up()`. All good!

File: src/Migrations/Version20240902163827.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    public function getDescription(): string
+    {
+        return 'Remove redundancy in new DonorAccount property names';
+    }
+
+    public function up(Schema $schema): void
+    {
+        $this->addSql(<<<'SQL'
+            ALTER TABLE DonorAccount
+                ADD homeAddressLine1 VARCHAR(255) DEFAULT NULL,
```
[the remark this answers is not in the document]
Comment (NoelLH) [reply src/Migrations/Version20240902163827.php:21]: https://github.com/thebiggive/matchbot/pull/992#pullrequestreview-2277042864

**Pull request merged**